File: code/src/test/java/chapter1/java/SessionTests.java
package chapter1.java;


import org.junit.jupiter.api.Test;

import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class SessionTests {

    private static Presenter alice = new Presenter("id1", "Alice");
    private static Presenter bob = new Presenter("id2", "Bob");
    private static final List<Presenter> noPresenters = Collections.emptyList();

    private Session aSession = new Session("The Title", List.of(bob, alice)); // <1>

    @Test
    public void can_change_title() {
        assertEquals(
            new Session("Another Title", List.of(bob, alice)),
            aSession.withTitle("Another Title")
        );
    }

    @Test
    public void can_change_presenters() {
        assertEquals(
            new Session("The Title", noPresenters),
            aSession.withPresenters(noPresenters)
        );
    }
}

File: code/src/test/kotlin/chapter1/kotlin/SessionTests.kt
package chapter1.kotlin

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class SessionTests {
    private val aSession = Session(
        "The Title",
        java.util.List.of(bob, alice)
    )

    @Test
    fun can_change_title() {
        assertEquals( // <1>
            Session(
                "Another Title",
                java.util.List.of(bob, alice)
            ),
            aSession.withTitle("Another Title")
        )
    }

    @Test
    fun can_change_presenters() {
        assertEquals(
            Session("The Title", noPresenters),
            aSession.withPresenters(noPresenters)
        )
    }

    companion object {
        private val alice = Presenter("id1", "Alice")
        private val bob = Presenter("id2", "Bob")
        private val noPresenters: List<Presenter> = emptyList()
    }
}